assets/scripts/streets/Geolocation.jsx

```diff
@@ -0,0 +1,156 @@
+/* global fetch */
+import React from 'react'
+import { connect } from 'react-redux'
+import { setSettings } from '../users/settings'
+import PropTypes from 'prop-types'
+import { Map, TileLayer, Marker, Popup } from 'react-leaflet'
+import SearchAddress from './SearchAddress'
+import {apiurlReverse, apikey} from './config'
+
+const OpenStreetMapTiles = 'http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png'
+const OpenStreetMapAttr = '&copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>'
+
+const zoomLevel = 12
+
+class Geolocation extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      mapCenter: [40.645, -73.975]
+    }
+
+    this.searchResults = this.searchResults.bind(this)
+    this.markerDrag = this.markerDrag.bind(this)
+    this.gotClick = this.gotClick.bind(this)
+    this.hidePopup = this.hidePopup.bind(this)
+  }
+
+/* start click event function */
+  gotClick (e) {
+    const displayAddressData = (res) => {
+      setSettings({ addressInformationLabel: res.features[0].properties.label })
+      setSettings({ markerLocation: res.features[0].geometry.coordinates.reverse() })
+
+      this.map.leafletElement.panTo(this.props.markerLocation)
+      console.log('Click Point Results : ', this.props.markerLocation, this.props.addressInformationLabel)
+    }
+
+    function gotAddressData (res, err) {
+      res.json().then(displayAddressData)
+    }
+
+    const options = {
+      lat: e.latlng.lat,
+      lng: e.latlng.lng
+    }
+
+    const clickUrl = `${apiurlReverse}?api_key=${apikey}&point.lat=${options.lat}&point.lon=${options.lng}`
+    fetch(clickUrl).then(gotAddressData)
+  }
+
+/* end click event function */
+
+/* start on marker drag function */
+  markerDrag (e) {
+    const gotJson = (res) => {
+      this.setState({
+        renderPopup: true
+      })
+
+      setSettings({ addressInformationLabel: res.features[0].properties.label })
+      setSettings({ markerLocation: e.target.getLatLng() })
+      console.log('Drag End Results : ', e.target.getLatLng(), this.props.addressInformationLabel)
+    }
+
+    function gotAddress (res, err) {
+      res.json().then(gotJson)
+    }
+
+    const targetCoords = e.target.getLatLng()
+
+    const dragEndUrl = `${apiurlReverse}?api_key=${apikey}&point.lat=${targetCoords.lat}&point.lon=${targetCoords.lng}`
+    fetch(dragEndUrl).then(gotAddress)
+  }
+
+/* end on marker drag function */
+
+// p = point , l = label
+  searchResults (p, l) {
```

Comment:
So it's easier to read later, just name the arguments `point` and `label` respectively instead of `p` and `l`, and you can then remove the comment.